internal/server/instance/drivers/driver_qemu.go

```diff
@@ -9307,3 +9322,40 @@ func (d *qemu) postCPUHotplug(monitor *qmp.Monitor) error {
 
 	return nil
 }
+
+// ConsoleLog returns all output sent to the instance's console's ring buffer since startup.
+func (d *qemu) ConsoleLog() (string, error) {
```

Comment:
We may want to use an `operationlock` for this one to prevent two concurrent attempts to fetch the console leading to a corrupted `console.log`.

This would need a new operation lock type to be added but that should still be pretty trivial.